### ISSUE TITLE
[LightGBM] [Fatal] Multiclass objective and metrics don't match

### ISSUE THREAD

**Issue by davidvilanova** [microsoft/LightGBM#3262]
Hi,
I have the following problem while trying to traing a model with three classes.

See reproducible example:

```
inTrain <- createDataPartition(iris$Species, p=.7, list = F)

train <- iris[inTrain,]
test <- iris[-inTrain,]

data_train <- lgb.Dataset(data = data.matrix(train[, 1:4]), label = train[, 5])
params <- list(objective = "multiclass", metric = "auc", num_class = 3)
model <- lgb.train(params,
                   data_train,
                   100,
                   min_data = 1,
                   learning_rate = .1)

Error in lgb.call("LGBM_DatasetCreateFromMat_R", ret = handle, private$raw_data,  : 
  [LightGBM] [Fatal] Multiclass objective and metrics don't match
```

What am i doing wrong ?

```
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Arch Linux

Matrix products: default
BLAS:   /usr/lib/libblas.so.3.9.0
LAPACK: /usr/lib/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] Matrix_1.2-18       lightgbm_2.3.2      R6_2.4.1            qwraps2_0.4.2      
 [5] reshape2_1.4.4      tibble_3.0.1        matrixStats_0.56.0  tidyr_1.1.0        
 [9] dplyr_1.0.0         caret_6.0-86        ggplot2_3.3.2       lattice_0.20-41    
[13] ROCR_1.0-11         randomForest_4.6-14 openxlsx_4.1.5      rmdformats_0.3.7   
[17] knitr_1.29         

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6         lubridate_1.7.9      class_7.3-17         utf8_1.1.4          
 [5] assertthat_0.2.1     digest_0.6.25        ipred_0.9-9          foreach_1.5.0       
 [9] plyr_1.8.6           stats4_4.0.2         evaluate_0.14        highr_0.8           
[13] pillar_1.4.4         rlang_0.4.6          rstudioapi_0.11      data.table_1.12.8   
[17] rpart_4.1-15         rmarkdown_2.3        labeling_0.3         splines_4.0.2       
[21] gower_0.2.1          stringr_1.4.0        munsell_0.5.0        compiler_4.0.2      
[25] xfun_0.15            pkgconfig_2.0.3      htmltools_0.5.0      nnet_7.3-14         
[29] tidyselect_1.1.0     prodlim_2019.11.13   bookdown_0.19        codetools_0.2-16    
[33] fansi_0.4.1          crayon_1.3.4         withr_2.2.0          MASS_7.3-51.6       
[37] recipes_0.1.13       ModelMetrics_1.2.2.2 grid_4.0.2           jsonlite_1.6.1      
[41] nlme_3.1-148         gtable_0.3.0         lifecycle_0.2.0      magrittr_1.5        
[45] pROC_1.16.2          scales_1.1.1         zip_2.0.4            cli_2.0.2           
[49] stringi_1.4.6        farver_2.0.3         timeDate_3043.102    ellipsis_0.3.1      
[53] generics_0.0.2       vctrs_0.3.1          lava_1.6.7           iterators_1.0.12    
[57] tools_4.0.2          glue_1.4.1           purrr_0.3.4          survival_3.2-3      
[61] yaml_2.2.1           colorspace_1.4-1 
```

**Comment by StrikerRUS** [comment]
@davidvilanova Hi!

For multiclass task you can use `auc_mu` metric: https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric-parameters.

**Comment by davidvilanova** [comment]
if i use `auc_mu`  i get the following error
```

Error in lgb.call("LGBM_BoosterCreate_R", ret = handle, train_set_handle,  : 
  [LightGBM] [Fatal] Label must be in [0, 3), but found 3 in label

Error in if (private$train_set$.__enclos_env__$private$version != private$train_set_version) { : 
  argument is of length zero
```

**Comment by StrikerRUS** [comment]
Class labels should start from 0.

https://github.com/microsoft/LightGBM/blob/d42f43132df63ae64783d3bc4f6366dd325116af/R-package/demo/multiclass_custom_objective.R#L6-L9
